### PR TITLE
feat(rule): report invalid gherkin file as lint problem

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Invalid gherkin file is now reported as lint problem

Screenshot from 2023-10-02 20-23-06

## Related Issue
- Part of https://github.com/gherlint/gherlint/issues/17

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests only (no source changes)
- [ ] Documentation only (no source changes)

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Documentation updated

### CODE REVIEW

File: tests/lib/linter/Linter.test.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        type: "error",
+        message: "Indentation error",
+        location: {},
+    },
+];
+
+describe("class: Linter", () => {
+    afterEach(() => {
+        jest.clearAllMocks();
+        jest.restoreAllMocks();
```
Comment (grgprarup) [inline tests/lib/linter/Linter.test.js:23]:
```suggestion
        jest.clearAllMocks();
        jest.restoreAllMocks();
```
Do we need both of them? or only one is sufficient?

File: tests/lib/linter/Linter.test.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    });
+    describe("method: lint", () => {
+        describe("without fix option", () => {
+            it("should return an empty array if no lint issues", () => {
+                const spyRunRules = jest
+                    .spyOn(Linter.prototype, "runRules")
+                    .mockReturnValue([]);
+
+                const linter = new Linter(config);
+                const result = linter.lint("");
```
Comment (grgprarup) [inline tests/lib/linter/Linter.test.js:33]:
```suggestion
                const linter = new Linter(config);
                const result = linter.lint("");
```
Frequent use of initiating `Linter`. Repetitive code lines, can't we optimize them?

**Review by grgprarup — APPROVED** [review]
Except for some queries, others LGTM. :+1:

File: tests/lib/linter/Linter.test.js (hunk 1 above)
Comment (saw-jan) [reply tests/lib/linter/Linter.test.js]: I overused it :sweat_smile:

File: tests/lib/linter/Linter.test.js (hunk 2 above)
Comment (saw-jan) [reply tests/lib/linter/Linter.test.js:32]: we could move it to hooks but different config requirement makes it complicated. There will be multiple hooks and might make grouping tests not easy

Comment (saw-jan) [reply tests/lib/linter/Linter.test.js:32]: But I agree, tests need some trimming

**Pull request merged**